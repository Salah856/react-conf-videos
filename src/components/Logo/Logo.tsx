import * as React from 'react';

import { Link } from 'react-router-dom';

import styles from './Logo.scss';

export const Logo: React.FunctionComponent<{}> = ({}) => (
  <Link
    to="/search"
    className={styles.logoLink}
    aria-label="Reactjs Videos Homepage"
  >
    <div className={styles.logo}>RV</div>
  </Link>
);
