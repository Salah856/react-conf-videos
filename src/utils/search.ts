import { compose, toLower, trim } from 'ramda'
import { remove as removeDiacritics } from 'diacritics'

export const cleanQuery = compose(
  removeDiacritics,
  toLower,
  trim
)

export const isFilterEmpty = (filterValue: string = '') => filterValue.trim() === ''
