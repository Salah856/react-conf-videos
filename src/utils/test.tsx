import { mount, shallow } from 'enzyme';
import * as React from 'react';
import { ReactElement } from 'react';
import { MemoryRouter } from 'react-router';
import { StateObservable } from 'redux-observable';
import { createMockStore } from 'redux-test-utils';
import { Subject } from 'rxjs';
import * as Immutable from 'seamless-immutable';

import { ApplicationState } from 'redux/modules';
import {
  Conference,
  IndexedConferences,
  IndexedPresenters,
  IndexedVideos,
  Presenter,
  Video
} from '../domain';

// cast to ApplicationState so TS doesn't complain when sending in partials to Epics for testing
export const mockStore = (state = {}) =>
  new StateObservable<ApplicationState>(
    new Subject(),
    state as ApplicationState
  );

const isStore = (store: any) => store.hasOwnProperty('isActionTypeDispatched');

// shallow mount component with a store in context
export const shallowWithStore = (storeP: any, component: ReactElement<{}>) => {
  const store = isStore(storeP) ? storeP : createMockStore(storeP);
  return shallow<any, any>(component, { context: { store } });
};

export const mountWithStore = (storeP: any, component: ReactElement<{}>) => {
  const store = isStore(storeP) ? storeP : createMockStore(storeP);
  return mount<any, any>(component, { context: { store } });
};

export const wrapWithMemoryRouter = (Component: React.ReactElement<any>) => (
  // key within location prop keeps changing, so manually set
  // so snapshorts are consistent
  <MemoryRouter initialEntries={[{ key: 'xxxx' }]}>{Component}</MemoryRouter>
);

/** Mock data */
export const mockVideo = (title = 'test title'): Video => ({
  link: 'a link',
  split: '',
  title: title || 'test title',
  length: '12:34',
  lightning: false,
  presenter: 'siomn carter',
  embeddableLink: 'a link'
});

export const mockVideos = (): IndexedVideos => ({
  aaa: mockVideo('aaa'),
  bbb: mockVideo('bbb'),
  ccc: mockVideo('ccc')
});

export const mockPresenter = (name = 'simon carter'): Presenter => ({
  name
});

export const mockPresenters = (): IndexedPresenters => ({
  xxx: mockPresenter(),
  zzz: mockPresenter('zzz')
});

export const mockConference = (): Conference => ({
  date: 'XX/YY/ZZZ',
  title: 'fake conf title',
  website: 'fake url',
  playlist: 'day 1',
  videos: ['aaa', 'bbb', 'ccc']
});

export const mockIndexedConferences = (): IndexedConferences => ({
  ['XXXX']: mockConference()
});

export const mockRouterState = (
  pathname = '/conference/8424e37df85b9eccbe48e9a55d93845e/react-conf-2018'
) => ({
  router: {
    location: {
      pathname,
      search: '',
      hash: ''
    }
  }
});

export const mockConferencePageSlice = () => ({
  conferencePage: {
    selectedConferenceId: 'XXXX',
    conference: mockConference()
  }
});

export const mockDataState = () =>
  Immutable({
    data: {
      conferences: mockIndexedConferences(),
      presenters: mockPresenters(),
      videos: mockVideos()
    }
  });
