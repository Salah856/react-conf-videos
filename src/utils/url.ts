import { compose, replace, toLower } from 'ramda';

// replaces white space with - and lower case
export const sluggifyUrl = compose(
  replace(/\s/g, '-'),
  toLower
);
