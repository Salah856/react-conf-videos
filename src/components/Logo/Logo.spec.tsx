import * as React from 'react'

import { mount } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { Logo } from './Logo'
import { wrapWithMemoryRouter } from 'utils/test'

describe('Logo', () => {
  it('should render', () => {
    // arrange
    // act
    const comp = mount(wrapWithMemoryRouter(<Logo />))

    // assert
    expect(toJSON(comp)).toMatchSnapshot()
  })
})
