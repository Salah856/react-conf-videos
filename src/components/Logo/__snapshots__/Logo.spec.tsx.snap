// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should render 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "xxxx",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "xxxx",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "xxxx",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Component>
      <Link
        aria-label="Reactjs Videos Homepage"
        className="logoLink"
        to="/search"
      >
        <a
          aria-label="Reactjs Videos Homepage"
          className="logoLink"
          href="/search"
          onClick={[Function]}
        >
          <div
            className="logo"
          >
            RV
          </div>
        </a>
      </Link>
    </Component>
  </Router>
</MemoryRouter>
`;
